src/dashboard/dashboard.js

```diff
@@ -1,10 +1,77 @@
 import React from "react";
+import ChatListComponent from "../chatlist/chatList";
+import { Button, withStyles } from "@material-ui/core";
+import styles from "./styles";
+import ChatViewComponent from "../chatview/chatView";
+import firebase from "../firebase";
 
 class DashboardComponent extends React.Component {
-    
-    render() {
-        return(<div>Hello World from Dashboard</div>)
-    }
+  constructor() {
+    super();
+    this.state = {
+      selectedChat: null,
+      newChatFormVisible: false,
+      email: null,
+      chats: [],
+    };
+  }
+
+  selectChat = (chatIndex) => {
+    this.setState({ selectedChat: chatIndex });
+  };
+
+  newChatBtnClicked = () => {
+    this.setState({ newChatFormVisible: true, selectedChat: null });
+  };
+
+  componentDidMount = () => {
+    firebase.auth().onAuthStateChanged(async (_usr) => {
+      if (!_usr) {
+        this.props.history.push("/login");
+      } else {
+        await firebase
+          .firestore()
+          .collection("chats")
+          .where("users", "array-contains", _usr.email)
+          .onSnapshot(async (res) => {
+            const chats = res.docs.map((_doc) => _doc.data());
+            await this.setState({
+              email: _usr.email,
+              chats: chats,
+            });
+            console.log(this.state);
+          });
+      }
+    });
+  };
+
+  signOut = () => firebase.auth().signOut();
```

Comment:
I can't believe how simple firebase makes things. I was going to ask you last week whether you were planning on adding a "sign out" button, and here it is - all the logic in one line!

Reply:
You can say that again! I am also amazed at how simple it is! Firebase has all necessary functions:)